test/integration/autonomous_upstream.cc

```diff
@@ -77,9 +77,13 @@ void AutonomousStream::sendResponse() {
 
   encodeHeaders(upstream_.responseHeaders(), headers_only_response);
   if (!headers_only_response) {
-    for (int32_t i = 0; i < response_data_blocks; ++i) {
-      encodeData(response_body_length,
-                 i == (response_data_blocks - 1) && !send_trailers && end_stream);
+    if (upstream_.responseBody().has_value()) {
+      encodeData(*upstream_.responseBody(), !send_trailers);
+    } else {
+      for (int32_t i = 0; i < response_data_blocks; ++i) {
+        encodeData(response_body_length,
```

Comment:
Should `headers_only_response` (line 71) also take the `upstream_.responseBody().has_value()` value into account?

Reply:
I think if you set headers-only response and set a body you're just doing it wrong =P